Library/src/Systems/Quota/Quota.cpp

```diff
@@ -0,0 +1,327 @@
+		}
+
+		if (UserTierResponse->HasTierName())
+		{
+			UserTierInfo.TierName = StringToTierNameEnum(UserTierResponse->GetTierName());
+		}
+	}
+}
+
+const FeatureQuotaInfo& FeatureQuotaResult::GetFeatureQuotaInfo() const
+{
+	return FeatureQuotaInfo;
+}
+
+void FeatureQuotaResult::OnResponse(const csp::services::ApiResponseBase* ApiResponse)
+{
+	ResultBase::OnResponse(ApiResponse);
+
+	auto* FeatureQuotaResponse			   = static_cast<chs::QuotaFeatureTierDto*>(ApiResponse->GetDto());
+	const csp::web::HttpResponse* Response = ApiResponse->GetResponse();
+
+	if (ApiResponse->GetResponseCode() == csp::services::EResponseCode::ResponseSuccess)
+	{
+		// Build the Dto from the response Json
+		FeatureQuotaResponse->FromJson(Response->GetPayload().GetContent());
+
+		if (FeatureQuotaResponse->HasFeatureName())
+		{
+			FeatureQuotaInfo.FeatureName = FeatureQuotaResponse->GetFeatureName();
+		}
+
+		if (FeatureQuotaResponse->HasTierName())
+		{
+			FeatureQuotaInfo.TierName = FeatureQuotaResponse->GetTierName();
+		}
+
+		if (FeatureQuotaResponse->HasLimit())
+		{
+			FeatureQuotaInfo.Limit = FeatureQuotaResponse->GetLimit();
+		}
+
+		if (FeatureQuotaResponse->HasPeriod())
+		{
+			FeatureQuotaInfo.Period = static_cast<PeriodEnum>(FeatureQuotaResponse->GetPeriod()->GetValue());
+		}
+
+		if (FeatureQuotaResponse->HasAllowReductions())
+		{
+			FeatureQuotaInfo.AllowReductions = FeatureQuotaResponse->GetAllowReductions();
+		}
+	}
+}
+
+const csp::common::Array<FeatureQuotaInfo>& FeaturesQuotaResult::GetFeaturesQuotaInfo() const
+{
+	return FeaturesQuotaInfo;
+}
+
+void FeaturesQuotaResult::OnResponse(const csp::services::ApiResponseBase* ApiResponse)
+{
+	ResultBase::OnResponse(ApiResponse);
+
+	auto* FeaturesQuotaResponse			   = static_cast<csp::services::DtoArray<chs::QuotaFeatureTierDto>*>(ApiResponse->GetDto());
+	const csp::web::HttpResponse* Response = ApiResponse->GetResponse();
+
+	if (ApiResponse->GetResponseCode() == csp::services::EResponseCode::ResponseSuccess)
+	{
+		// Build the Dto from the response Json
+		FeaturesQuotaResponse->FromJson(Response->GetPayload().GetContent());
+		std::vector<chs::QuotaFeatureTierDto>& QuotaArray = FeaturesQuotaResponse->GetArray();
+
+		for (size_t i = 0; i < QuotaArray.size(); ++i)
+		{
+			if (QuotaArray[i].HasFeatureName())
+			{
+				FeaturesQuotaInfo[i].FeatureName = QuotaArray[i].GetFeatureName();
+			}
+
+			if (QuotaArray[i].HasTierName())
+			{
+				FeaturesQuotaInfo[i].TierName = QuotaArray[i].GetTierName();
+			}
+
+			if (QuotaArray[i].HasLimit())
+			{
+				FeaturesQuotaInfo[i].Limit = QuotaArray[i].GetLimit();
+			}
+
+			if (QuotaArray[i].HasPeriod())
+			{
+				FeaturesQuotaInfo[i].Period = static_cast<PeriodEnum>(QuotaArray[i].GetPeriod()->GetValue());
+			}
+
+			if (QuotaArray[i].HasAllowReductions())
+			{
+				FeaturesQuotaInfo[i].AllowReductions = QuotaArray[i].GetAllowReductions();
+			}
+		}
+	}
+}
+const csp::common::String& TierNameEnumToString(TierNames Value)
```

Comment:
The tier names have changed. They are captured in the PRD:
basic
premium
pro
enterprise